File: app/src/main/java/com/prography/prography_androidstudy/Room/UserDatabase.java
package com.prography.prography_androidstudy.Room;

import android.content.Context;

import androidx.room.Database;
import androidx.room.Room;
import androidx.room.RoomDatabase;


@Database(entities = {User.class}, version = 1)
public abstract class UserDatabase extends RoomDatabase {

    public static volatile UserDatabase INSTANCE;

    static UserDatabase getDatabase(final Context context) {
        if (INSTANCE == null) {
            synchronized (UserDatabase.class) {
                if (INSTANCE == null) {
                    // Create Database here
                    INSTANCE = Room.databaseBuilder(context.getApplicationContext(), UserDatabase.class, "users")
                            .build();
                }
            }
        }
        return INSTANCE;
    }

    abstract public UserDao userDao();
}

File: app/src/main/java/com/prography/prography_androidstudy/Room/UserDatabase.kt
package com.prography.prography_androidstudy.Room

import android.content.Context
import androidx.room.Database
import androidx.room.Room
import androidx.room.RoomDatabase

@Database(entities = [User::class], version = 1)
abstract class UserDatabase : RoomDatabase() {

    abstract val userDao: UserDao

    companion object {
        private var instance: UserDatabase? = null

        @Synchronized
        fun getInstance(context: Context): UserDatabase {
            if (instance == null) {
                instance = Room.databaseBuilder(context.applicationContext, UserDatabase::class.java, "user.db")
                    .fallbackToDestructiveMigration()
                    .build()
            }
            return instance as UserDatabase
        }
    }
}